KILTprotocol/mashnet-node
# feat: add launch token claiming process

## fixes KILTProtocol/ticket#1172
Adds `pallet_vesting` and custom `kilt_launch` pallet which enable
* Enable setting vesting and KILT custom balance lock in the Genesis config
* Add extrinsic for the claiming process of migrating tokens, vesting and locks from an unowned Genesis-block address to a user-owned account address
  * either from a single source or a vector of sources to a single target address
* Add extrinsic to change transfer account
* Add extrinsic to force_unlock
* Add extrinsic to transfer KILT custom locked tokens

## ToDo

- [x] Update unit tests to make use of `vest` which was made public in c54b7a4
- [x] Benchmarks

## How to test:

It can also be tested in the Polkadot Apps but I suppose it makes more sense if I walk you through this. 

In short: During the genesis block building, the transfer account and a bunch of pseudo accounts (whose addresses etc. are described in <you_know_where>) for vesting and locks are set up. For each of those pseudo accounts, you need to call `migrate_genesis_account` or `migrate_multiple_genesis_accounts` signed by the transfer account. Then, either vesting or the custom lock should be set up. 

```
cargo test --all --all-features
```

## Checklist:
